### PR TITLE
[SW-570] Multisig: Propose transaction (UI)

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/composables/maxAmount.ts [hunk 1]
```diff
@@ -82,6 +82,9 @@ export function useMaxAmount({ store, formModel }: MaxAmountOptions) {
         selectedAssetDecimals.value = val.selectedAsset.decimals!;
       }
 
+      const numericAmount = (val.amount && val.amount > 0) ? val.amount : 0;
```
Comment (konradodo) [inline src/composables/maxAmount.ts:85]:
```suggestion
      const numericAmount = (val?.amount > 0) ? val.amount : 0;
```

File: src/popup/components/Modals/TransferSend.vue [hunk 2]
```diff
@@ -66,92 +82,105 @@ export default {
...
-    if (this.tokenContractId && this.availableTokens[this.tokenContractId]) {
-      this.transferData.selectedAsset = this.availableTokens[this.tokenContractId];
+  setup(props, { root }) {
+    const currentRenderedComponent = ref<Vue.Component>();
+    const currentStep = ref<Step>(STEPS.form);
+    const error = ref(false);
+    const transferData = ref<TransferFormModel>({});
+
+    const availableTokens = useState<ITokenList>('fungibleTokens', 'availableTokens');
+    const isConnected = useGetter('isConnected');
```
Comment (konradodo) [inline src/popup/components/Modals/TransferSend.vue:94]:
```suggestion
    const isConnected = useGetter<boolean>('isConnected');
```

File: src/popup/components/Modals/TransferSend.vue [hunk 3]
```diff
@@ -66,92 +82,105 @@ export default {
...
+    function handleReviewSuccess() {
+      closeModal();
+    }
+
+    function editTransfer() {
+      error.value = false;
+      currentStep.value = STEPS.form;
+    }
+
+    const steps: Record<Step, { component: any, onSuccess: any }> = {
```
Comment (konradodo) [inline src/popup/components/Modals/TransferSend.vue:140]:
```suggestion
   
    const steps: Record<Step, { component: Vue.Component, onSuccess: any }> = {
```

File: src/popup/components/TransferReview.vue [hunk 4]
```diff
@@ -279,7 +309,15 @@ export default defineComponent({
       }
     }
 
-    async function submit() {
+    // TODO
+    function proposeMultisigTransaction() {
+      root.$store.dispatch('modals/open', {
```
Comment (konradodo) [inline src/popup/components/TransferReview.vue:314]: if my pr will be merged before that yours,  then please use useDIspatch

File: src/popup/components/TransferReview.vue [hunk 5]
```diff
@@ -288,30 +326,34 @@ export default defineComponent({
...
+            note,
+          });
+        case props.isMultisig:
+          return proposeMultisigTransaction();
+        default:
+          return transfer({
+            amount,
+            recipient,
+            selectedAsset,
+          });
```
Comment (konradodo) [inline src/popup/components/TransferReview.vue:351]:
```suggestion
      if (props.isAddressUrl) {
          return sendTip({
            amount,
            recipient,
            selectedAsset,
            note,
          });
          }
       
        if (props.isMultisig) {
          return proposeMultisigTransaction();   
       }
       
          return transfer({
            amount,
            recipient,
            selectedAsset,
          });
```

File: src/popup/pages/Dashboard.vue [hunk 6]
```diff
@@ -143,6 +145,12 @@ export default {
         name: MODAL_TRANSFER_SEND,
       });
     },
+    openMultisigTransactionProposalModal() {
+      this.$store.dispatch('modals/open', {
```
Comment (konradodo) [inline src/popup/pages/Dashboard.vue:149]: if you can, useDispatch

File: src/popup/pages/Dashboard.vue [hunk 7]
```diff
@@ -22,6 +22,8 @@
         />
       </div>
 
+      <a @click="openMultisigTransactionProposalModal()">Transaction proposal</a>
```
Comment (konradodo) [inline src/popup/pages/Dashboard.vue:25]: this is not a link, more semantic is option is use button

Comment (konradodo) [reply src/popup/pages/Dashboard.vue:25]: Transaction proposal -> to translation

File: src/composables/maxAmount.ts (hunk 1 above)
Comment (peronczyk) [reply src/composables/maxAmount.ts:85]: Nope, you can't check if null is bigger than a number.

File: src/popup/pages/Dashboard.vue (hunk 7 above)
Comment (peronczyk) [reply src/popup/pages/Dashboard.vue:25]: This is temporary thing. I added TODO comment.

File: src/popup/pages/Dashboard.vue (hunk 6 above)
Comment (peronczyk) [reply src/popup/pages/Dashboard.vue:149]: This is options API. No way of using composables.

File: src/popup/components/TransferReview.vue (hunk 5 above)
Comment (peronczyk) [reply src/popup/components/TransferReview.vue:351]: OK

File: src/popup/components/TransferReview.vue (hunk 4 above)
Comment (peronczyk) [reply src/popup/components/TransferReview.vue:314]: Sure. But it's not merged :)

File: src/popup/components/Modals/TransferSend.vue (hunk 3 above)
Comment (peronczyk) [reply src/popup/components/Modals/TransferSend.vue:140]: Sure, makes sense.

**Review by konradodo — APPROVED** [review]

File: src/popup/components/SelectableListItem.vue [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+export default defineComponent({
+  components: {
+    BtnPlain,
+  },
+  props: {
+    selected: Boolean,
+  },
+  // setup() {
+  // },
```
Comment (pstachel) [inline src/popup/components/SelectableListItem.vue:24]: Should be removed

**Review by pstachel — APPROVED** [review]

File: src/popup/components/SelectableListItem.vue (hunk 8 above)
Comment (peronczyk) [reply src/popup/components/SelectableListItem.vue:24]: Sure

**Review by CedrikNikita — CHANGES_REQUESTED** [review]
Clicking on add payload results in 
```
TypeError: Cannot read properties of undefined (reading 'length')
```

There are no `payload` in the type `TransferFormModel`

**Comment by peronczyk** [comment]
> Clicking on add payload results in
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'length')
> ```

Nice finding. It was caused by the fact that the payload was required when opening the payload modal. But it shouldn't, because when opening for the first time there is no payload. Now the payload defaults to empty string.

**Review by CedrikNikita — APPROVED** [review]
Code changes look reasonable.

**Pull request merged**